test/unittests/execute_numeric_test.cpp

```diff
@@ -481,6 +481,12 @@ TEST(execute_numeric, i64_extend_i32_s)
     EXPECT_THAT(i64_extend_i32_s(0x80000001_u32), Result(0xffffffff80000001_u64));
     EXPECT_THAT(i64_extend_i32_s(0xfffffffe_u32), Result(0xfffffffffffffffe_u64));
     EXPECT_THAT(i64_extend_i32_s(0xffffffff_u32), Result(0xffffffffffffffff_u64));
+
+    // Put some garbage in the Value's high bits.
```

Comment:
Hmm, what about a case where `i64` has top bit set, but `i32` does not have top bit set. That should clear the top bits.

Reply:
Done.